tests/python/grad_test.py

```diff
@@ -3,6 +3,7 @@
 
 
 def grad_test(tifunc, npfunc=None):
```

Comment:
I believe this test will not be executed and should be removed. The same function already exists as a helper function in `test_ad_basics.py`.